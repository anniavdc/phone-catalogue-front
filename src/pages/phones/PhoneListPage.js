import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from "react-router-dom";

// files
import ProductCard from './../../components/productCard/ProductCard';
import { phoneList } from './../../store/actions/index';

export default function PhoneListPage({ match }) {
    const phones = useSelector(state => state.phones);
    const dispatch = useDispatch();
    let history = useHistory();

    useEffect(() => {
        dispatch(phoneList());
    }, [dispatch]);

    const handleProductClick = (id) => {
        history.push(`${match.path}/${id}`);
    };

    const baseURL = process.env.REACT_APP_API_BASE;
    return (
        <React.Fragment>
            <h2 className="products-title">Phone Catalogue</h2>
            <div className="products-wrapper">
                {phones.phoneList.map(phone =>
                    <ProductCard
                        key={phone.id}
                        title={phone.name}
                        content={phone.manufacturer}
                        info={`${phone.price}â‚¬`}
                        imageUrl={`${baseURL}/public/${phone.imageFileName}`}
                        onCardClick={() => handleProductClick(phone.id)}
                    />
                )}
            </div>
        </React.Fragment>
    )
}
