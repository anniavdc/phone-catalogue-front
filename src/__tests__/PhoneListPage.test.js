import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

// files
import render from '../utils/tests/renderWithRouter';
import PhoneListPage from '../pages/phones/PhoneListPage';
import { fireEvent } from '@testing-library/dom';

const initialState = {
    phones: {
        phoneList: [
            {
                color: "black",
                description: "This is iPhone 7.",
                id: 0,
                imageFileName: "IPhone_7.png",
                manufacturer: "Apple",
                name: "iPhone 7",
                price: 769,
                processor: "A10 Fusion",
                ram: 2,
                screen: "4,7 inch IPS"
            }
        ]
    }
};

const mockStore = configureStore([thunk]);

describe('PhoneListPage', () => {
    let store;
    beforeEach(() => {
        store = mockStore(initialState);
    })

    test('render', async () => {
        const { container } = render(
            <Provider store={store}>
                <PhoneListPage match={{ params: { }, isExact: true, path: "/app", url: "/app" }} />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });

    test('it should redirect to detail', async () => {
        const { container, history } = render(
            <Provider store={store}>
                <PhoneListPage match={{ params: { }, isExact: true, path: "/app", url: "/app" }} />
            </Provider>,
            "/app",
            { wrapper: MemoryRouter }
        );
        const productCard = container.querySelector('.product-card');
        fireEvent.click(productCard);

        await expect(history.location.pathname).toMatch(/0/);
    });

})
