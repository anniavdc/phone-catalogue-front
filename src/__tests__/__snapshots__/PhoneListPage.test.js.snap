// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneListPage render 1`] = `
<div>
  <h2
    class="products-title"
  >
    Phone Catalogue
  </h2>
  <div
    class="products-wrapper"
  >
    <div
      class="product-card"
    >
      <p
        class="card-info"
      >
        769â‚¬
      </p>
      <div
        class="card-image"
        style="background-image: url(http://localhost:5000/public/IPhone_7.png);"
      />
      <div>
        <h3
          class="card-title"
        >
          iPhone 7
        </h3>
        <p
          class="card-content"
        >
          Apple
        </p>
      </div>
    </div>
  </div>
</div>
`;
